spring-cloud-kubernetes-integration-tests/run.sh

```diff
@@ -116,17 +116,33 @@ main() {
 	#"${ISTIOCTL}" install --set profile=demo
 
 	# running tests..
-	for p in "${INTEGRATION_PROJECTS[@]}"; do
+	if [[ $CIRCLECI ]]; then
+		#This splits projects across all circleci instances, it returns a list of projects separated by a space
+		SPLIT_PROJECTS=$(printf "%s\n" "${INTEGRATION_PROJECTS[@]}" | circleci tests split)
+		SPLIT_PROJECTS=$(echo $SPLIT_PROJECTS | sed 's/ /,/g')
+		echo "split tests $SPLIT_PROJECTS"
+		#This splits the projects back into an array so we can iterate over them
+		IFS=',' read -ra PROJECTS <<< "$SPLIT_PROJECTS"
+		echo "projects $PROJECTS"
```

Comment:
should this be `echo "projects ${PROJECTS[@]}"`? I mean `echo "projects $PROJECTS"` will show only the first one in the array.

But why the need for `IFS=',' read -ra PROJECTS <<< "$SPLIT_PROJECTS"`? We already have this array as `INTEGRATION_PROJECTS`. I mean if you do : 

```
	echo "${PROJECTS[@]}"
	echo "********"
	echo "${INTEGRATION_PROJECTS[@]}"

```

you will get the same output in both arrays. Unless I am missing something, we can drop creating the `PROJECTS` array.

Reply:
As an addendum, I am also wondering why we would not split this by timings? `--split-by=timings --timings-type=classname`... I guess it's because there is no place to save the results to, like via that `store_test_results`, right?

Reply:
`PROJECTS` are the split `INTEGRATION_PROJECTS` its a subset.
We could use timing data, but there are not a large amount of integration tests here and its almost 1-2 tests per instance at the moment, so splitting by timing is probably not going to buy us anything.